test_pull_requests

```diff
@@ -733,23 +935,25 @@ popd
       return comment
     end
 
-    # get_comment_with_prefix returns a comment authored by the bot with the given prefix, if one exists.
-    # In order to ensure that we don't return a comment that starts with a longer prefix that aliases the
-    # prefix we're passed, the regex we use requires a word boundary anchor, whitespace character or line
-    # end at the end of the prefix we're passed.
-    def get_comment_with_prefix(issue_id, repo, comment_prefix, comments=nil)
-      prefix_comment = nil
+    # get_comment_matching_regex returns a comment authored by the bot matching the supplied regex, if one exists
+    def get_comment_matching_regex(issue_id, repo, pattern, comments=nil)
+      matching_comment = nil
 
       comments = comments ? comments : get_comments(issue_id, repo)
 
       comments.each do |comment|
-        if comment['body'] =~ /^#{Regexp.quote(comment_prefix)}(\b|$|\s)/ && (comment['user']['login'] == Properties['bot_github_user'])
-          prefix_comment = comment
+        if comment['body'] =~ pattern && (comment['user']['login'] == Properties['bot_github_user'])
+          matching_comment = comment
           break
         end
       end
 
-      prefix_comment
+      matching_comment
+    end
+
+    # get_comment_with_prefix returns a comment authored by the bot with the given prefix, if one exists
```

Comment:
it looks right to me? I'll tweak the formatting and fix the comment for the subsequent function though.